allocations/tasks.py

```diff
@@ -348,28 +350,47 @@ def check_keycloak_consistency():
     active_groups = {
         group.get("name")
         for group in groups._client.get(url=groups_url, briefRepresentation=False)
+        # Filter for any groups with have `has_active_allocation` == "true"
         if "true" in group.get("attributes", {}).get("has_active_allocation", [])
     }
     inactive_groups = {
         group.get("name")
         for group in groups._client.get(url=groups_url, briefRepresentation=False)
-        if "false" in group.get("attributes", {}).get("has_active_allocation", [])
+        # Filter for any groups with have `has_active_allocation` == "false" or None
+        if any(
+            inactive in group.get("attributes", {}).get("has_active_allocation", [])
+            for inactive in ("false", None)
+        )
     }
 
     LOG.info(f"CONSISTENCY: {len(active_groups)} active groups in Keycloak")
     LOG.info(f"CONSISTENCY: {len(inactive_groups)} inactive groups in Keycloak")
 
     for project in active_projects:
-        if project not in active_groups:
+        if project in inactive_groups:
```

Comment:
@super-cooper 
Just to comment, this would be cleaner if e.g. both `active_projects` and `inactive_groups` were `set()`s, rather than 1 set and one dict, so we could use set notation for the intersections.

active_in_portal_inactive_in_keycloak = active_projects-active_groups
inactive_in_portal_active_in_keycloak = inactive_projects-inactive_groups

Reply:
They all appear to be set comprehensions, so they could be combined as such.